### ISSUE TITLE
Don't know how to put java.util.List<Model> in a Bundle

### ISSUE THREAD

**Issue by IlyaEremin** [sockeqwe/fragmentargs#10]
Hello!
I have `Model` that implements `Parcelable`. Now, I want to pass `List<Model>` as argument, but fragmentArgs forces me to use `ArrayList<Model>` as parameter. I see message: `Don't know how to put java.util.List<Model> in a Bundle`

I want to keep my argument as `List`. How can I resolve this case?

Can you provide new annotation like `@ParceAs(ArrayList.class)` or summat?

**Comment by sockeqwe** [comment]
Im afraid, but as you already know there is no method to put a `List<T>` in a `Bundle`. 
Your suggestion with `@ParceAs`:
Why should `@ParceAs(Long.class)` not work for a `Date` objects. But there are to many possibilities to consider.

However, I did something similar in another AnnotationProcessor of mine called [ParcelablePlease](https://github.com/sockeqwe/ParcelablePlease) by providing Baggers. Check the Bagger section. Do you think something similar like this would solve you problem.

My idea would be to provide something like this:
```java
@Arg(bundler = MyModelListBundler.class) 
```

where you have to implement `MyModelListBundler´ by yourself, because only you as developer know what kind of `List` implementation you are using:

```java
public class MyModelListBunder implements Bundler<List<Model>> {

      @Override
      public void put(String key, List<Model> value,  Bundle bundle) {
             // cast to ArrayList, because we know it's an array list
             bundle.putParcelableArrayList(key, (ArrayList<Model>) value); 

            // Alternatively if we don't use ArrayList
            ArrayList<Model> casted = new ArrayList<Model>(value);
            bundle.putParcelableArrayList(key, casted);

       }
      
      @Override
      public List<Model> get(String key, Bundle bundle) {
           return bundle.getParcelableArrayList(key);
      }
}
```

What do you think?

**Comment by IlyaEremin** [comment]
Oh, I think it's a great approach.
It will allow us not only using all kinds of lists, but also wrap in bundle any other object!

**Comment by sockeqwe** [comment]
alright, i will add that in Version 2.1 and probably provide some default "Bundler" implementations.

Im a little bit busier than I wanted to, but I think Version 2.1 will be released at the begin of march

**Comment by sockeqwe** [comment]
Implemented, ships in version 2.1 which is released
